### PR TITLE
Current Transaction Sankey Diagram

### PR DESCRIPTION
**Pull request by umarluqman**
Screenshot at Dec 21 09-04-04

- The inputs can be chosen by user and it will reflect in the sankey diagram.
- Transaction size is hardcoded for now
- The mining fee is also hardcoded
- It is inside canvas and user need to drag it to the center currently as it will render a bit off the center.

This is just an initial visualization of the current transaction, we will iterate on this later on.

### CODE REVIEW

**Comment by umarluqman** [comment]
after merge from master

**Comment by umarluqman** [comment]
Screenshot at Dec 21 11-32-55
final result

File: apps/mobile/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "bdk-rn": "^0.30.0",
     "bip39": "^3.1.0",
     "d3": "^7.9.0",
+    "d3-sankey": "^0.12.3",
```
Comment (Psycarlo) [inline apps/mobile/package.json:24]: Let's add this dependency in the docs page `apps/docs/pages/getting-started/dependencies.mdx`

File: apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx [hunk 2]
```diff
@@ -218,6 +304,22 @@ export default function IOPreview() {
...
       </SSMainLayout>
+      <View style={{ position: 'absolute', flex: 1, top: 100 }}>
+        <GestureDetector gesture={panGesture}>
+          <Animated.View
+            style={[
+              { width: sankeyWidth, height: sankeyHeight },
+              animatedStyle
+            ]}
+          >
+            <UtxoFlow
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx:315]: this should be called SSUtxoFlow

File: apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx [hunk 3]
```diff
@@ -147,46 +272,7 @@ export default function IOPreview() {
...
-                </SSHStack>
-                <SSText color="muted" size="xxs">
-                  "{output.label}"
-                </SSText>
-              </SSVStack>
-            ))}
-          </SSVStack>
-        </SSHStack>
-        <SSVStack>
+        <SSVStack style={styles.bottomSection}>
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx:275]: We can also leave the style in-line in this case

File: apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx [hunk 4]
```diff
@@ -72,16 +84,129 @@ export default function IOPreview() {
...
+  const canvasSize = { width: GRAPH_WIDTH * 1.5, height: GRAPH_HEIGHT }
+  const sankeyWidth = canvasSize.width
+  const sankeyHeight = canvasSize.height - 200
+
+  const translateX = useSharedValue(-width * 0.4)
+  const translateY = useSharedValue(0)
+  const savedTranslateX = useSharedValue(-width * 0.4)
+  const savedTranslateY = useSharedValue(0)
+
+  const panGesture = Gesture.Pan()
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx:100]: I wonder if we would have the sankey diagram in multiple pages, this code should be moved inside the component or a wrapper component in the future

Comment (umarluqman) [reply apps/mobile/app/(authenticated)/account/[id]/signAndSend/ioPreview.tsx:100]: the SSUtxoFlow is the diagram, we can just reuse it if we want to show the sankey diagram. The gesture IMHO should be outside of the SSUtxoFlow component. 

yeah the naming is a bit confusing, I will rename it to SSSankeyDiagram

**Comment by Psycarlo** [comment]
Please use _function_ keyword form components

**Comment by umarluqman** [comment]
> Please use _function_ keyword form components

I added it as eslint rule as well

**Review by Psycarlo — APPROVED** [review]

**Pull request merged**